File: src-tauri/src/types/system.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HealthStatus {
    pub healthy: bool,
    pub version: Option<String>,
    pub error: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SystemInfo {
    pub smolvm_version: Option<String>,
    pub smolvm_path: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SmolvmBinary {
    pub path: String,
    pub env: Vec<(String, String)>,
    pub cwd: Option<String>,
    pub prefix_args: Vec<String>,
    pub arg_join: Option<String>,
}

File: src/lib/types/system.ts
export interface HealthStatus {
  healthy: boolean;
  version: string | null;
  error: string | null;
}

export interface SystemInfo {
  smolvm_version: string | null;
  smolvm_path: string | null;
}

export interface SmolvmBinary {
  path: string;
  env: [string, string][];
  cwd: string | null;
  prefix_args: string[];
  arg_join: string | null;
}

export interface MachineStats {
  name: string;
  pid: number | null;
  cpu_percent: number;
  memory_bytes: number;
  timestamp: number;
  alive: boolean;
}

export interface SystemStats {
  per_machine: MachineStats[];
  total_cpu_percent: number;
  total_memory_bytes: number;
  host_memory_total_bytes: number;
  host_cpu_count: number;
  timestamp: number;
}

export type View =
  | "machines"
  | "newMachine"
  | "images"
  | "volumes"
  | "packs"
  | "stats"
  | "settings";
